spec/puck/dependency_resolver_spec.rb

```diff
@@ -105,6 +105,27 @@ module Puck
         end
 
         it 'does not write a bundle configuration' do
+          resolved_gem_dependencies # force evaluation
+          bundler_config = Pathname.new(app_dir_path).join('.bundle', 'config')
+          bundler_config.should_not exist
+        end
+      end
+
+      context 'with custom app_dir' do
```

Comment:
It would give you, the tester, more control. While I'm aware that it isn't the common case, unit tests are supposed to excercise specific parts of the code, rather than simulating real-world usage. Using explicit options rather than having the instance under test derive them from the environment is, in my opinion, much clearer and isolates whatever is tested from external dependencies better.

Having that said, it is still crucial to test the current directory solution, as that is the most common case.